src/Bicep.Core/Emit/ParameterAssignmentEvaluator.cs

```diff
@@ -161,17 +167,20 @@ SemanticModel bicepModel
             return Result.For(DiagnosticBuilder.ForPosition(import.DeclaringImportedSymbolsListItem.OriginalSymbolName).TypeSymbolUsedAsValue(import.Name));
         });
 
-        private Result EvaluateImportedVariable(SemanticModel importedFrom, ImportedSymbol symbol)
+        private static Template GetTemplate(SemanticModel model)
         {
-            if (importedFrom.Root.VariableDeclarations
-                .Where(v => importedFrom.Exports.ContainsKey(v.Name))
-                .Where(v => LanguageConstants.IdentifierComparer.Equals(v.Name, symbol.OriginalSymbolName))
-                .FirstOrDefault() is VariableSymbol exportedVariable)
+            var textWriter = new StringWriter();
+            using var writer = new SourceAwareJsonTextWriter(model.FileResolver, textWriter)
             {
-                return bicepEvaluators.GetOrAdd(importedFrom, model => new(model)).EvaluateVariable(exportedVariable);
-            }
+                // don't close the textWriter when writer is disposed
+                CloseOutput = false,
+                Formatting = Formatting.Indented
+            };
+            var (template, _) = new TemplateWriter(model).GetTemplate(writer);
 
-            return Result.For(DiagnosticBuilder.ForPosition(symbol.DeclaringImportedSymbolsListItem.OriginalSymbolName).ImportedSymbolNotFound(symbol.OriginalSymbolName ?? LanguageConstants.MissingName));
+            TemplateEngine.ProcessTemplateLanguageExpressions(null, null, null, template, TemplateWriter.NestedDeploymentResourceApiVersion, null);
```

Comment:
[nit]: I _think_ this step can be skipped, since the compiled template will still be queried via a `TemplateVariablesEvaluator` that evaluates expressions as needed. But it also doesn't do any harm.

Reply:
I tried removing this and the scenario test started failing - leaving in for now